### ISSUE TITLE
[Question] Binding to two buttons?

### ISSUE THREAD

**Issue by MetalSpork** [Ryochan7/DS4Windows#526]
I wanted to make it so holding down one of the D-Pad directions would also have the Right Trigger held down as well. However the macro functionality will not work for this purpose because the buttons need to be held for as long as I have the D-Pad direction held down. 

Is there something I'm missing, or an alternate way of doing this?

**Comment by tremendoza** [comment]
InputMapper can do what you're asking with its macro feature set. I recommend 1.6.10.

**Comment by MetalSpork** [comment]
> InputMapper can do what you're asking with its macro feature set. I recommend 1.6.10.

Unfortunately InputMapper is not compatible with Cemuhook. Ds4Windows and ScpToolkit are the only options.

**Comment by Ryochan7** [comment]
There is no way to make that type of mapping work with DS4Windows. It is possible with InputMapper 1.6.10 although it requires more work than it should. I am not sure about IM 1.7. Neither version can do proper stick axis scaling though.

**Comment by MetalSpork** [comment]
Well, that sucks. I will have to resort to AutoHotkey shenanigans.

**Comment by Ryochan7** [comment]
I cannot get macro mapping working in IM 1.7 with the most recent build.

I would hope to implement some changes to the mapping system to allow multiple outputs for each input in the future.

**Comment by MetalSpork** [comment]
Alternatively, what about a 'Wait for Button Release' event in the DS4Windows macro system? Or a way to execute one half of a macro on button down and the other half on button up?

**Comment by MetalSpork** [comment]
Another implementation idea: Pause until Button Release. Essentially a special pause that will increment for the same length of time that its trigger is held down. Sort of similar to the Wait for Button Release idea above but more compatible with the way it currently works.

**Comment by Ryochan7** [comment]
Having actions executed on a button release should be handled in some form in the future. I would rather get rid of the general idea of macros and just make some sort of linked action sequence being bound to an input control. I don't know if something like the old slot based system that I wrote in the past or using something like the activator system used in Steam would work out better.

**Comment by Aidolii** [comment]
Was this issue closed because of this recent change?
```
Fixed remapping multiple controls to one XInput output control
Fixed remapping of XInput controls
```
Because the first one sounds like it could be related. I couldn't figure out how to do it though.

**Comment by Ryochan7** [comment]
This issue was closed since there was nothing more to discuss. This type of feature is still desired but it will not get implemented in the current code base.

The message in the changelog covers the opposite problem with different DS4 inputs being bound to a single XInput control. I tried to refactor mapping with version 1.6.9 to get rid of making new Dictionary objects during the mapping process. That change busted remapping in many ways. Version 1.6.10 fixed problems introduce with version 1.6.9.
